packages/satusehat/test/Adaptor.test.js

```diff
@@ -0,0 +1,523 @@
+import { expect } from 'chai';
+import { enableMockClient } from '@openfn/language-common/util';
+
+import { execute, get, post, put, patch } from '../src/Adaptor.js';
+
+// This creates a mock client which acts like a fake server.
+// It enables pattern-matching on the request object and custom responses
+// For the full mock API see
+// https://undici.nodejs.org/#/docs/api/MockPool?id=mockpoolinterceptoptions
+const baseUrl = 'https://fake.satusehat.server.com';
+const testServer = enableMockClient(baseUrl);
+
+describe('execute', () => {
+  beforeEach(() => {
+    testServer
+      .intercept({
+        path: '/oauth2/v1/accesstoken?grant_type=client_credentials',
+        method: 'POST',
+      })
+      .reply(200, {
+        access_token: 'fake-token',
+      });
+  });
+  it('executes each operation in sequence', async () => {
+    const state = {
+      configuration: {
+        baseUrl,
+        clientId: 'someclientid',
+        clientSecret: 'someclientsecret',
+      },
+    };
+    const operations = [
+      state => {
+        return { counter: 1 };
+      },
+      state => {
+        return { counter: 2 };
+      },
+      state => {
+        return { counter: 3 };
+      },
+    ];
+
+    const finalState = await execute(...operations)(state);
+
+    expect(finalState).to.eql({ counter: 3 });
+  });
+
+  it('assigns references and data to the initialState', async () => {
+    const state = {
+      configuration: {
+        baseUrl,
+        clientId: 'someclientid',
+        clientSecret: 'someclientsecret',
+      },
+    };
+
+    const finalState = await execute()(state);
+
+    expect(finalState).to.eql({
+      configuration: {
+        baseUrl: 'https://fake.satusehat.server.com',
+        clientId: 'someclientid',
+        clientSecret: 'someclientsecret',
+        access_token: 'fake-token',
+      },
+      references: [],
+      data: null,
+    });
+  });
+});
+
+describe('getOrganizations', () => {
+  beforeEach(() => {
+    testServer
```

Comment:
You shouldn't need to `.persist()` AND `beforeEach()`. 

You should just be able to set up a persistent interceptor at the top of the file (perhaps in a `before()` block) and use it in all the tests